Bottom: 3ff660268308f81ffa10068086e46e1eb0d84b77
Top:    897cee0b17f17050fe3667cbaf7c6d9d3f66ab9d
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-22 15:36:51 -0700

Refresh of dbg-fork

---

diff --git a/src/preload.c b/src/preload.c
index fba0281..d1d8b06 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -413,9 +413,13 @@ int socket(int domain, int type, int protocol)
 	if (index < 0)
 		return index;
 
+	if ((domain == PF_INET || domain == PF_INET6) &&
+	    (type == SOCK_STREAM) && (!protocol || protocol == IPPROTO_TCP) && fork_support) {
+		printf("skipping rsocket call\n");
+		goto realsock;
+	}
+
 	recursive = 1;
-printf("skipping rsocket call\n");
-goto realsock;
 	ret = rsocket(domain, type, protocol);
 	recursive = 0;
 	if (ret >= 0) {
